experiments/tse.py

```diff
@@ -1,3 +1,120 @@
 """Turbo spin echo sequence."""
+# %%
+# imports
+import logging
+import numpy as np
+import matplotlib.pyplot as plt
+from console.spcm_control.interface_acquisition_parameter import AcquisitionParameter, Dimensions
+from console.spcm_control.acquisition_control import AcquistionControl
+from console.spcm_control.interface_acquisition_data import AcquisitionData
+from console.utilities.plot_unrolled_sequence import plot_unrolled_sequence
+import console.utilities.sequences as sequences
+from scipy.signal import decimate
 
-# TODO: Implementation
\ No newline at end of file
+# %%
+# Create acquisition control instance
+configuration = "../device_config.yaml"
+acq = AcquistionControl(configuration_file=configuration, console_log_level=logging.INFO, file_log_level=logging.DEBUG)
+
+# %%
+# Construct sequence
+seq, traj = sequences.tse.tse_2d.constructor(
+    echo_time=20e-3,
+    repetition_time=300e-3,
+    etl=1,
+    gradient_correction=510e-6,
+    rf_duration=200e-6,
+    ro_bandwidth=20e3,
+    fov=Dimensions(x=220e-3, y=220e-3, z=225e-3),
+    # n_enc=Dimensions(x=64, y=64, z=0)
+    n_enc=Dimensions(x=128, y=128, z=0)
+)
+
+# %%
+# Plot sequence section
+acq.seq_provider.from_pypulseq(seq)
+seq_unrolled = acq.seq_provider.unroll_sequence(larmor_freq=2e6, grad_offset=Dimensions(0, 0, 0))
+fig, ax = plot_unrolled_sequence(seq_unrolled, seq_range=(0, 640000), output_limits=[200, 6000, 6000, 6000])
```

Comment:
Is it possible to read function parameters from the current settings? Maybe todo?

Reply:
I'm not sure I understand what you mean, this is just to plot the unrolled sequence. We could seperate unrolling and execution of a sequence. Then one could set a sequence in the acquisition control and plot exactly the same sequence that is replayed?

Reply:
In this part: ...output_limits=[200, 6000, 6000, 6000], it seemed to me we unnecessarly set the output limits again. Is not it already set in the project settings globally? Do we need to specify them again? I could not follow this part.

Reply:
This is only to plot the unrolled sequence with the correct output limits. The unrolled sequence is in int16 values.
We could add the plot function to the sequence provider, then we don't need to pass them again.